homeassistant/components/alarm_control_panel/group.py

```diff
@@ -0,0 +1,236 @@
+    DOMAIN, PLATFORM_SCHEMA, SERVICE_ALARM_DISARM, SERVICE_ALARM_ARM_HOME,
+    SERVICE_ALARM_ARM_AWAY, SERVICE_ALARM_ARM_NIGHT, SERVICE_ALARM_TRIGGER,
+    SERVICE_ALARM_ARM_CUSTOM_BYPASS)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_PRE_PENDING_STATE, ATTR_POST_PENDING_STATE,
+    CONF_ENTITIES, CONF_NAME, EVENT_HOMEASSISTANT_START,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS, STATE_ALARM_PENDING,
+    STATE_ALARM_TRIGGERED)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change
+
+ATTR_TRIGGERED_PANELS = 'triggered_panels'
+
+CONF_CODE_FORMAT = 'code_format'
+
+DEFAULT_ALARM_NAME = 'Group Alarm'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_ALARM_NAME): cv.string,
+    vol.Optional(CONF_CODE_FORMAT, default=''): cv.string,
+    vol.Required(CONF_ENTITIES): vol.All(cv.ensure_list, [cv.entity_id]),
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the group alarm platform."""
+    add_devices([GroupAlarm(
+        hass,
+        config[CONF_NAME],
+        config[CONF_CODE_FORMAT],
+        config[CONF_ENTITIES]
+        )])
+
+
+class GroupAlarm(alarm.AlarmControlPanel):
+    """Implement services and compute state for the group alarm platform."""
+
+    def __init__(self, hass, name, code_format, entities):
+        """Initialize the platform."""
+        self._hass = hass
+        self._name = name
+        self._code_format = code_format if code_format else None
+        self._entity_ids = entities
+        self._triggered_panels = []
+        self._state = None
+        self._pre_state = self._state
+        self._post_state = self._state
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def group_alarm_state_listener(entity, old_state, new_state):
+            """Handle device state changes."""
+            self.async_schedule_update_ha_state(True)
+
+        @callback
+        def group_alarm_startup(event):
+            """Update template on startup."""
+            async_track_state_change(
+                self.hass, self._entity_ids, group_alarm_state_listener)
+
+            self.async_schedule_update_ha_state(True)
+
+        self.hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_START, group_alarm_startup)
+
+    @asyncio.coroutine
+    def _async_send_message(self, service, **kwargs):
+        """Send message to all entities in the group."""
+        payload = {key: val for key, val in kwargs.items() if val}
+
+        tasks = []
+        for entity_id in self._entity_ids:
+            sending_payload = deepcopy(payload.copy())
+            sending_payload[ATTR_ENTITY_ID] = entity_id
+            tasks.append(self.hass.services.async_call(
+                DOMAIN, service, sending_payload))
```

Comment:
You need to specify `blocking=True`. Also the entities will automatically be expanded for you, so no need for a for-loop I think.

https://github.com/home-assistant/home-assistant/blob/ca5f4709564773c8cebd6ee40aa7cc1095b19105/homeassistant/components/light/group.py#L188-L189